### PR TITLE
Fixvecmul

### PR DESCRIPTION
**Pull request by btracey**
VecMul was not working in the default case. This fixes and adds test.

### CODE REVIEW

**Comment by btracey** [comment]
PTAL @kortschak

File: mat64/matrix_test.go [hunk 1]
```diff
@@ -37,6 +38,23 @@ func panics(fn func()) (panicked bool, message string) {
 	return
 }
 
+// outside is a type that is a Matrix which does not implement other interfaces
+type outside blas64.General
```
Comment (kortschak) [inline mat64/matrix_test.go]: Don't we already have a testing type like this?

Comment (btracey) [inline mat64/matrix_test.go]: Oh, do we? I don't remember seeing it anywhere, but maybe I just forgot.

File: mat64/vector.go [hunk 2]
```diff
@@ -254,12 +260,28 @@ func (v *Vector) MulVec(a Matrix, trans bool, b *Vector) {
...
-			w.mat.Data[r*v.mat.Inc] = f
+			w.mat.Data[r*w.mat.Inc] = f
 		}
 	}
 	*v = w
 }
 
+// Equals compares the vectors represented by b and the receiver and returns true
+// if the vectors are element-wise equal.
+func (v *Vector) Equals(b *Vector) bool {
```
Comment (kortschak) [inline mat64/vector.go]: EqualsVec?

File: mat64/matrix_test.go (hunk 1 above)
Comment (kortschak) [inline mat64/matrix_test.go]: I thought you put it in.

File: mat64/vector.go [hunk 3]
```diff
@@ -236,15 +236,21 @@ func (v *Vector) MulVec(a Matrix, trans bool, b *Vector) {
 			t = blas.Trans
 		}
 		blas64.Gemv(t, 1, amat, b.mat, 0, w.mat)
-	case Vectorer:
```
Comment (kortschak) [inline mat64/vector.go]: Do we need to lose this case?

File: mat64/matrix_test.go (hunk 1 above)
Comment (kortschak) [inline mat64/matrix_test.go]: https://github.com/gonum/matrix/blob/master/mat64/dense_test.go#L20
https://github.com/gonum/matrix/blob/master/mat64/mul_test.go#L221

The type is yours.

Comment (btracey) [inline mat64/matrix_test.go]: Oh. Sorry. Thanks for the catch.

Comment (btracey) [inline mat64/matrix_test.go]: Fixed.

File: mat64/vector.go (hunk 2 above)
Comment (btracey) [inline mat64/vector.go]: Done.

File: mat64/vector.go (hunk 3 above)
Comment (btracey) [inline mat64/vector.go]: Put back in.

**Comment by btracey** [comment]
PTAL

**Comment by kortschak** [comment]
LGTM

**Pull request merged**